tmt/steps/execute/__init__.py

```diff
@@ -420,14 +409,40 @@ def _map_old_methods(self) -> None:
         # shell.tmt, beakerlib.tmt ---> tmt
         how = 'tmt'
         self.warn(f"Use 'how: {how}' in the execute step instead (L2).")
-        self.data[0]['how'] = how
+        data[0].how = how
         # Store shell or beakerlib as the default test framework
         # (used when the framework is not defined in the L1 metadata)
         framework = matched.group(1)
         self.warn(f"Set 'framework: {framework}' in test metadata (L1).")
         self._framework = framework
         self.warn("Support for old methods will be dropped in tmt-2.0.")
 
+    def _normalize_data(self, raw_data: List[tmt.steps._RawStepData]) -> List[tmt.steps.StepData]:
+        data = super()._normalize_data(raw_data)
+
+        # Map old methods now if there is no run (and thus no wake up)
+        # TODO: if not self.plan.my_run:
+        self._map_old_methods(data)
```

Comment:
@psss what was the purpose of this condition? I changed the code to call `_map_old_methods()` always when converting raw data to step data instance, without this the respective tests failed.

Reply:
I believe the extra hacks for the old execution methods needed to be applied just once:

* either around `init` when there's no run, this would be used for `tmt plan show`
* or during `wake` after command line options are applied, which could override `data`

Anyway, in #1490 I'm proposing to completely remove the old hacks, so hopefully this part will soon be gone.

Reply:
We should be fine then, after my refactoring, the method is applied to raw data from fmf (or dicts), and `wake()` no longer needs to apply it.